src/main/java/org/openrewrite/java/testing/mockito/ReplaceInitMockToOpenMock.java

```diff
@@ -0,0 +1,149 @@
+
+                    private boolean isAnnotatedMethodPresent(J.ClassDeclaration cd, AnnotationMatcher beforeEachMatcher) {
+                        return cd.getBody().getStatements().stream().anyMatch(
+                                st -> st instanceof J.MethodDeclaration &&
+                                        ((J.MethodDeclaration) st).getLeadingAnnotations().stream().anyMatch(beforeEachMatcher::matches)
+                        );
+                    }
+
+                    final TreeVisitor<J, ExecutionContext> updateJUnitLifecycleMethods = new JavaIsoVisitor<ExecutionContext>() {
+
+                        @Override
+                        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration cd, ExecutionContext ctx) {
+                            if (!isAnnotatedMethodPresent(cd, AFTER_EACH_MATCHER) && isAnnotatedMethodPresent(cd, BEFORE_EACH_MATCHER)) {
+                                maybeAddImport("org.junit.jupiter.api.AfterEach");
+                                cd = JavaTemplate.builder("@AfterEach\nvoid tearDown() throws Exception {\n}")
+                                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "junit-jupiter-api-5"))
+                                        .imports("org.junit.jupiter.api.AfterEach")
+                                        .build()
+                                        .apply(getCursor(), cd.getBody().getCoordinates().lastStatement());
+                            }
+
+                            cd = super.visitClassDeclaration(cd, ctx);
+                            return autoFormat(cd, ctx);
+                        }
+
+                        @Override
+                        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                            J.MethodDeclaration md = super.visitMethodDeclaration(method, ctx);
+
+                            if (service(AnnotationService.class).matches(updateCursor(md), BEFORE_EACH_MATCHER) && md.getBody() != null) {
+                                maybeRemoveImport("org.mockito.MockitoAnnotations.initMocks");
+                                for (Statement st : md.getBody().getStatements()) {
+                                    if (st instanceof J.MethodInvocation && INIT_MOCKS_MATCHER.matches((J.MethodInvocation) st)) {
+                                        return JavaTemplate.builder(variableName + " = MockitoAnnotations.openMocks(this);")
+                                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "mockito-core"))
+                                                .imports("org.mockito.MockitoAnnotations")
+                                                .contextSensitive()
+                                                .build()
+                                                .apply(getCursor(), st.getCoordinates().replace());
+                                    }
```

Comment:
When stepping through with the debugger I saw the second edit cycle had the same printed output, but a different `initMocks` instance (_not_ `openMocks`) on these lines: https://github.com/openrewrite/rewrite/blob/9569cfb6b7c88d015ba1369fa5639bcf23db4dc0/rewrite-test/src/main/java/org/openrewrite/test/LargeSourceSetCheckingExpectedCycles.java#L91-L98

I suspect this JavaTemplate has something to do with that, as the `st.getCoordinates().replace()` when visiting the method declaration is a bit suspect.